packages/quantic/force-app/main/default/lwc/quanticTooltip/quanticTooltip.js

```diff
@@ -30,23 +38,127 @@ export default class QuanticTooltip extends LightningElement {
    * Method tha hides the tooltip.
    */
   @api hideTooltip() {
+    this.resetPosition();
     this.isVisible = false;
   }
 
+  /** @type {boolean} */
+  isInitialLoad = true;
+  /** @type {NodeJS.Timeout} */
+  resizeTimer;
+  /** @type {boolean} */
+  displayTooltipAboveTarget = true;
   /** @type {boolean} */
   isVisible = false;
 
+  connectedCallback() {
+    window.addEventListener('resize', this.handleResizeAndScroll);
+    window.addEventListener('scroll', this.handleResizeAndScroll);
```

Comment:
I'm curious why aren't all the calculations actually done only when we show the tooltip?

Like before showing the tooltip, calculate all the things, then add the class to make it visible?

Reply:
That's a good question actually cause this was one of the main problems I encountered. 
My initial idea was like you described, do the calculation when we are about to show the tooltip but then got a serious issue kinda hard to imagine but I will do my best x)

basically this is about moving the tooltip  in the y axis, to show it above the target or below it.
Now imagine we are doing the calculations  when we are about to show the tooltip and we are in a situation where the tooltip needs to be displayed `below`.

That's what will happen:

- Initially the tooltip is hidden and is above the target.
- Step01: The user hovers over the target.
- We execute the `showTooltip` method which is responsible for doing the calculation and displaying the tooltip.
- We do the calculation, we found that the tooltip needs to be displayed below so we apply a translate the tooltip in the y axis.
- while this transition is happening we display the tooltip.
- at a certain fraction of second while doing this y axis translation and since the tooltip will be transition from above to below, at a certain fraction of second the tooltip will cover the target, this will yield a `mouseleave` event on the target cause technically the mouse is no longer on the target now that it is covered.
- on `mouseleaver` the target calls `hideTooltip` method.
- `hideTooltip` method resets the position of the tooltip
- we come back to Step 01...

This was the infinite loop that was occuring, that's why I switched to doing the calculation related to the y axis on resize and scroll and not on hover, so the tooltip is already on its correct y position when the user hovers.

Reply:
Lol so basically you're telling me we would only need to add a delay before displaying the tooltip corresponding to the transition time to support the case where we would only do the calculations at the moment where we want to display it?

I think I would prefer that than having something that constantly calculates ALL tooltips positions while scrolling etc. no?

Because right now for each instance of a tooltip in the page you're adding a scroll and resize listener, IMO that's pretty costly.

I would just update the showTooltip to add a timer to only show after the transition is finished.
Or is it possible to simply not have a transition on the translation? The transition of the tooltip imo only should be on the opacity right?

Reply:
Of course it would be better to re-calculate on each resize/scroll, let me just try it again  cause I'm kinda sure this wasn't the only issue that made change this idea.

I will it try it again and intensively test and if it works as expected it would be better indeed.